### PR TITLE
Add support for multiple solution types in PCS CLI

### PR DESCRIPTION
**Pull request by jillcary**
# Type of change? 

- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 
In preparation for adding a new ARM template for local development and for future PCS solution types, this change does the following:
- Moves the existing templates into the solutions folder under remotemonitoring
- Adds a new sku flag for local development (replacing the testing flag)

**Checklist:**

- [x] All tests passed
- [x] The code follows the code style and conventions of this project
- [x] The change requires a change to the documentation
- [x] I have updated the documentation accordingly

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -64,51 +64,56 @@ let answers: Answers = {};
 
 const program = new Command(packageJson.name)
     .version(packageJson.version, '-v, --version')
-    .option('-t, --type <type>', 'Solution Type: remotemonitoring', /^(remotemonitoring|test)$/i, 'remotemonitoring')
-    .option('-s, --sku <sku>', 'SKU Type: basic, standard, or test', /^(basic|standard|test)$/i, 'basic')
+    .option('-t, --type <type>', 'Solution Type: remotemonitoring',
+            /^(remotemonitoring|test)$/i,
+            'remotemonitoring')
+    .option('-s, --sku <sku>', 'SKU Type (only for Remote Monitoring): basic, standard, or local', /^(basic|standard|local)$/i, 'basic')
```
[the remark this answers is not in the document]
Comment (jillcary) [reply src/index.ts]: Updated to match

File: package.json [hunk 2]
```diff
@@ -13,8 +13,9 @@
   ],
   "main": "./publish/index.js",
   "scripts": {
-    "start": "npm run build && npm test && npm run copy",
+    "start": "npm run build && npm test && npm run copy && npm run copysolutions",
     "copy": "copyfiles ./remotemonitoring/**/*.* ./publish/",
```
Comment (ppathan) [inline package.json]: I would update copy to use solutions folder to publish and remove copysolutions

File: src/deploymentmanager.ts [hunk 3]
```diff
@@ -82,183 +83,195 @@ export class DeploymentManager implements IDeploymentManager {
...
-                this._sku += '-static-map';
-            } else {
-                resources.forEach((resource: ResourceModels.GenericResource) => {
-                    if (resource.plan && resource.plan.name && resource.plan.name.toLowerCase() === 'internal1') {
-                        freeBingMapResourceCount++;
+        return this._client.resources.list({ filter: 'resourceType eq \'Microsoft.BingMaps/mapApis\'' })
+            .then((resources: ResourceModels.ResourceListResult) => {
+
+                if (this._solutionType === 'remotemonitoring') {
+
```
Comment (ppathan) [inline src/deploymentmanager.ts]: nit: remove extra line

File: src/deploymentmanager.ts [hunk 4]
```diff
@@ -82,183 +83,195 @@ export class DeploymentManager implements IDeploymentManager {
...
-            // using static map for China environment by default since Bing Map resource is not available.
-            if (this._options.environment && this._options.environment.name === AzureEnvironment.AzureChina.name) {
-                this._sku += '-static-map';
-            } else {
-                resources.forEach((resource: ResourceModels.GenericResource) => {
-                    if (resource.plan && resource.plan.name && resource.plan.name.toLowerCase() === 'internal1') {
-                        freeBingMapResourceCount++;
+        return this._client.resources.list({ filter: 'resourceType eq \'Microsoft.BingMaps/mapApis\'' })
+            .then((resources: ResourceModels.ResourceListResult) => {
+
```
Comment (ppathan) [inline src/deploymentmanager.ts]: nit: remove extra line

File: src/deploymentmanager.ts [hunk 5]
```diff
@@ -82,183 +83,195 @@ export class DeploymentManager implements IDeploymentManager {
...
+                        if (freeBingMapResourceCount >= MAX_BING_MAP_APIS_FOR_INTERNAL1_PLAN) {
+                            this._sku += '-static-map';
+                        }
                     }
-                });
-                if (freeBingMapResourceCount >= MAX_BING_MAP_APIS_FOR_INTERNAL1_PLAN) {
-                    this._sku += '-static-map';
+
+                    this._template = require(armTemplatePath + this._sku + '.json');
+
```
Comment (ppathan) [inline src/deploymentmanager.ts]: nit: remove extra line

**Pull request merged**